
From: Kaigai Kohei <kaigai@ak.jp.nec.com>

This patch declares atomic_add_return() as an alias of __atomic_add(). 
atomic64_add_return(),atomic_sub_return() and atomic64_sub_return() are
same.

Signed-off-by: KaiGai, Kohei <kaigai@ak.jp.nec.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/asm-sparc64/atomic.h |    6 ++++++
 1 files changed, 6 insertions(+)

diff -puN include/asm-sparc64/atomic.h~atomic_inc_return-for-sparc64 include/asm-sparc64/atomic.h
--- 25/include/asm-sparc64/atomic.h~atomic_inc_return-for-sparc64	2004-09-11 16:08:35.827656208 -0700
+++ 25-akpm/include/asm-sparc64/atomic.h	2004-09-11 16:08:35.830655752 -0700
@@ -40,6 +40,12 @@ extern int __atomic64_sub(__s64, atomic6
 #define atomic_inc_return(v) __atomic_add(1, v)
 #define atomic64_inc_return(v) __atomic64_add(1, v)
 
+#define atomic_sub_return(i, v) __atomic_sub(i, v)
+#define atomic64_sub_return(i, v) __atomic64_sub(i, v)
+
+#define atomic_add_return(i, v) __atomic_add(i, v)
+#define atomic64_add_return(i, v) __atomic64_add(i, v)
+
 /*
  * atomic_inc_and_test - increment and test
  * @v: pointer of type atomic_t
_
